lib/crypto.js

```diff
@@ -305,10 +305,22 @@ Sign.prototype.sign = function sign(options, encoding) {
   var passphrase = options.passphrase || null;
 
   // Options specific to RSA
-  var rsaPadding = options.padding || constants.RSA_PKCS1_PADDING;
-  var pssSaltLength = constants.RSA_PSS_SALTLEN_MAX_SIGN;
-  if (typeof options.saltLength === 'number') {
-    pssSaltLength = options.saltLength;
+  var rsaPadding = constants.RSA_PKCS1_PADDING;
+  if (options.hasOwnProperty('padding')) {
```

Comment:
Maybe just check for `options.padding !== undefined` instead of using `hasOwnProperty`? The latter fails when `options` doesn’t inherit from `Object`, and I think there’s no reason to require `padding` to be a *own* property of `options`.